src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -0,0 +1,367 @@
+    }
+
+    /**
+     * Is first tour step
+     */
+    public get onFirstStep(): boolean {
+        return this._onFirstStep;
+    }
+
+    /* Show resize message */
+    public get onResizeMessage(): boolean {
+        return this._onResizeMessage;
+    }
+
+    /* Current tour step number */
+    public get currentTourStepDisplay(): number {
+        return this._currentTourStepIndex + 1;
+    }
+
+    /* Total count of tour steps */
+    public get currentTourStepCount(): any {
+        return this._currentTour && this._currentTour.steps ? this._currentTour.steps.length : 0;
+    }
+
+    /* Prevents the tour from advancing by clicking the backdrop */
+    public get preventBackdropFromAdvancing(): boolean {
+        if (this._currentTour) {
+            return this._currentTour && (this._currentTour.preventBackdropFromAdvancing ? this._currentTour.preventBackdropFromAdvancing : false);
+        }
+        return false;
+    }
+
+    /**
+     * Get the tour step with defined orientation
+     * @param index current tour step index
+     */
+    private getPreparedTourStep(index: number): TourStep | undefined {
+        if (this._currentTour) {
+            return this.setTourOrientation(this._currentTour.steps[index]);
+        } else {
+            return undefined;
+        }
+    }
+
+    /**
+     * Set orientation of the passed on tour step
+     * @param step  passed on tour step of a guided tour
+     */
+    private setTourOrientation(step: TourStep): TourStep {
+        const convertedStep = cloneDeep(step);
+        if (convertedStep.orientation && !(typeof convertedStep.orientation === 'string') && (convertedStep.orientation as OrientationConfiguration[]).length) {
+            (convertedStep.orientation as OrientationConfiguration[]).sort((a: OrientationConfiguration, b: OrientationConfiguration) => {
+                if (!b.maximumSize) {
+                    return 1;
+                }
+                if (!a.maximumSize) {
+                    return -1;
+                }
+                return b.maximumSize - a.maximumSize;
+            });
+
+            let currentOrientation: Orientation = Orientation.Top;
+            (convertedStep.orientation as OrientationConfiguration[]).forEach((orientationConfig: OrientationConfiguration) => {
+                if (!orientationConfig.maximumSize || window.innerWidth <= orientationConfig.maximumSize) {
+                    currentOrientation = orientationConfig.orientationDirection;
+                }
+            });
+
+            convertedStep.orientation = currentOrientation;
+        }
+        return convertedStep;
+    }
+
+    /**
+     * Send a GET request for the guided tour settings of the current user
+     */
+    private findGuidedTourSettings(): Observable<GuidedTourSettings | undefined> {
+        return this.http.get<GuidedTourSettings>(this.resourceUrl, { observe: 'response' }).map((res: EntityResponseType) => {
+            if (res.body) {
```

Comment:
This would mean that if res.body is null or undefined, map returns undefined.
This is not good! Better throw an error or just filter in a pipe.